### DISCUSSION TITLE
Locking a key in the entire cliuster to perform an operation then unlocking it

### DISCUSSION THREAD

**Discussion by ghost** [infinispan/infinispan#12609] [Q&A] [answered]
Hello Infinispan community,

i have a replicated cache of protostream data type that contains an array in its value, i want to lock the key to do operations on the array and then unlock the key cluster wide, and while it is locked i do not want any other operations to be allowed until the lock is released, is such feature supported by infinispan hotrod client ?

**Comment by pruivo** [comment, marked as answer]
No, it is not.

**Reply by ghost** [reply]
would using infinispan-clustered-lock to create locks to control threads over a cluster work ?

**Reply by pruivo** [reply]
the clustered locks are only available for library mode only. It will work if you want to create a cluster in your application.

**Comment by tristantarrant** [comment]
Instead of locking, you should use versioning as it provides the best performance.
Do something like this:
```java
do {
    MetadataValue<MyClass> mv = cache.getWithMetadata(key);
    MyClass v = mv.getValue(); 
    // Manipulate v
} while (!cache.replaceWithVersion(key, v, mv.getVersion()));
```
